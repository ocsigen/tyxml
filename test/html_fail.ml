open Tyxml.Html ;;

div [a [a []]] ;;

div [address [a [address []]]];;

body [dl [dd [txt "foo"]; dt [a [h1 [txt "foo"]]]]];;
