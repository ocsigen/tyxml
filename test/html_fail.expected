[1mLine 2, characters 5-13[0m:
2 | div [a [a []]] ;;
         [1;31m^^^^^^^^[0m
[1;31mError[0m: This expression has type
         ([> ([> 'c Html_types.a ] as 'b) Html_types.a ] as 'a)
         Tyxml.Html.elt = 'a Tyxml_html.elt
       but an expression was expected of type
         ([< Html_types.div_content_fun ] as 'd) Tyxml.Html.elt =
           'd Tyxml_html.elt
       Type 'a = [> `A of 'b ] is not compatible with type
         'd =
           [< `A of Html_types.flow5_without_interactive_header_footer
            | `Abbr
            | `Address
            | `Article
            | `Aside
            | `Audio of Html_types.flow5_without_media_header_footer
            | `Audio_interactive of
                Html_types.flow5_without_media_header_footer
            | `B
            | `Bdo
            | `Blockquote
            | `Br
            | `Button
            | `Canvas of Html_types.flow5_without_header_footer
            | `Cite
            | `Code
            | `Command
            | `Datalist
            | `Del of Html_types.flow5_without_header_footer
            | `Details
            | `Dfn
            | `Div
            | `Dl
            | `Em
            | `Embed
            | `Fieldset
            | `Figure
            | `Footer
            | `Form
            | `H1
            | `H2
            | `H3
            | `H4
            | `H5
            | `H6
            | `Header
            | `Hgroup
            | `Hr
            | `I
            | `Iframe
            | `Img
            | `Img_interactive
            | `Input
            | `Ins of Html_types.flow5_without_header_footer
            | `Kbd
            | `Keygen
            | `Label
            | `Main
            | `Map of Html_types.flow5_without_header_footer
            | `Mark
            | `Menu
            | `Meter
            | `Nav
            | `Noscript of Html_types.flow5_without_noscript_header_footer
            | `Object of Html_types.flow5_without_header_footer
            | `Object_interactive of Html_types.flow5_without_header_footer
            | `Ol
            | `Output
            | `P
            | `PCDATA
            | `Picture
            | `Pre
            | `Progress
            | `Q
            | `Ruby
            | `Samp
            | `Script
            | `Section
            | `Select
            | `Small
            | `Span
            | `Strong
            | `Style
            | `Sub
            | `Sup
            | `Svg
            | `Table
            | `Template
            | `Textarea
            | `Time
            | `U
            | `Ul
            | `Var
            | `Video of Html_types.flow5_without_media_header_footer
            | `Video_interactive of
                Html_types.flow5_without_media_header_footer
            | `Wbr ] 
       Type 'b = [> `A of 'c ] is not compatible with type
         Html_types.flow5_without_interactive_header_footer 
       Types for tag `A are incompatible
[1mLine 2, characters 14-28[0m:
2 | div [address [a [address []]]];;
                  [1;31m^^^^^^^^^^^^^^[0m
[1;31mError[0m: This expression has type
         ([> ([> Html_types.address ] as 'b) Html_types.a ] as 'a)
         Tyxml.Html.elt = 'a Tyxml_html.elt
       but an expression was expected of type
         ([< Html_types.address_content_fun ] as 'c) Tyxml.Html.elt =
           'c Tyxml_html.elt
       Type 'a = [> `A of 'b ] is not compatible with type
         'c =
           [< `A of
                Html_types.flow5_without_interactive_sectioning_heading_header_footer_address
            | `Abbr
            | `Audio of
                Html_types.flow5_without_media_sectioning_heading_header_footer_address
            | `Audio_interactive of
                Html_types.flow5_without_media_sectioning_heading_header_footer_address
            | `B
            | `Bdo
            | `Blockquote
            | `Br
            | `Button
            | `Canvas of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Cite
            | `Code
            | `Command
            | `Datalist
            | `Del of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Details
            | `Dfn
            | `Div
            | `Dl
            | `Em
            | `Embed
            | `Fieldset
            | `Figure
            | `Form
            | `Hr
            | `I
            | `Iframe
            | `Img
            | `Img_interactive
            | `Input
            | `Ins of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Kbd
            | `Keygen
            | `Label
            | `Main
            | `Map of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Mark
            | `Menu
            | `Meter
            | `Noscript of
                Html_types.flow5_without_noscript_sectioning_heading_header_footer_address
            | `Object of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Object_interactive of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Ol
            | `Output
            | `P
            | `PCDATA
            | `Picture
            | `Pre
            | `Progress
            | `Q
            | `Ruby
            | `Samp
            | `Script
            | `Select
            | `Small
            | `Span
            | `Strong
            | `Style
            | `Sub
            | `Sup
            | `Svg
            | `Table
            | `Template
            | `Textarea
            | `Time
            | `U
            | `Ul
            | `Var
            | `Video of
                Html_types.flow5_without_media_sectioning_heading_header_footer_address
            | `Video_interactive of
                Html_types.flow5_without_media_sectioning_heading_header_footer_address
            | `Wbr ] 
       Type 'b = [> `Address ] is not compatible with type
         Html_types.flow5_without_interactive_sectioning_heading_header_footer_address 
       Types for tag `A are incompatible
[1mLine 2, characters 30-48[0m:
2 | body [dl [dd [txt "foo"]; dt [a [h1 [txt "foo"]]]]];;
                                  [1;31m^^^^^^^^^^^^^^^^^^[0m
[1;31mError[0m: This expression has type
         ([> ([> Html_types.h1 ] as 'b) Html_types.a ] as 'a) Tyxml.Html.elt
           = 'a Tyxml_html.elt
       but an expression was expected of type
         ([< Html_types.dt_content_fun ] as 'c) Tyxml.Html.elt =
           'c Tyxml_html.elt
       Type 'a = [> `A of 'b ] is not compatible with type
         'c =
           [< `A of
                Html_types.flow5_without_interactive_sectioning_heading_header_footer_address
            | `Abbr
            | `Audio of
                Html_types.flow5_without_media_sectioning_heading_header_footer_address
            | `Audio_interactive of
                Html_types.flow5_without_media_sectioning_heading_header_footer_address
            | `B
            | `Bdo
            | `Blockquote
            | `Br
            | `Button
            | `Canvas of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Cite
            | `Code
            | `Command
            | `Datalist
            | `Del of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Details
            | `Dfn
            | `Div
            | `Dl
            | `Em
            | `Embed
            | `Fieldset
            | `Figure
            | `Form
            | `Hr
            | `I
            | `Iframe
            | `Img
            | `Img_interactive
            | `Input
            | `Ins of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Kbd
            | `Keygen
            | `Label
            | `Main
            | `Map of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Mark
            | `Menu
            | `Meter
            | `Noscript of
                Html_types.flow5_without_noscript_sectioning_heading_header_footer_address
            | `Object of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Object_interactive of
                Html_types.flow5_without_sectioning_heading_header_footer_address
            | `Ol
            | `Output
            | `P
            | `PCDATA
            | `Picture
            | `Pre
            | `Progress
            | `Q
            | `Ruby
            | `Samp
            | `Script
            | `Select
            | `Small
            | `Span
            | `Strong
            | `Style
            | `Sub
            | `Sup
            | `Svg
            | `Table
            | `Template
            | `Textarea
            | `Time
            | `U
            | `Ul
            | `Var
            | `Video of
                Html_types.flow5_without_media_sectioning_heading_header_footer_address
            | `Video_interactive of
                Html_types.flow5_without_media_sectioning_heading_header_footer_address
            | `Wbr ] 
       Type 'b = [> `H1 ] is not compatible with type
         Html_types.flow5_without_interactive_sectioning_heading_header_footer_address 
       Types for tag `A are incompatible

